pg/src/main/java/org/bouncycastle/openpgp/PGPPublicKey.java

```diff
@@ -921,7 +914,7 @@ private static PGPPublicKey removeCert(
         PGPPublicKey    returnKey = new PGPPublicKey(key);
         boolean         found = false;
 
-        for (int i = 0; i < returnKey.ids.size(); i++)
```

Comment:
Iterating the list forwards and removing from it can cause `ConcurrentModificationExceptions`. Therefore I reversed the direction.